src/sentry/utils/appleconnect/itunes_connect.py

```diff
@@ -302,9 +302,26 @@ def _request_trusted_phone_info(self) -> None:
             },
             timeout=REQUEST_TIMEOUT,
         )
-        if response.status_code != HTTPStatus.OK:
-            raise ITunesError(f"Unexpected response status: {response.status_code}")
-        info = response.json()["trustedPhoneNumber"]
+        # Fastlane actually doesn't check the response status in their implementation and simply
+        # checks whether the response returns the expected fields associated with 2FA. This is
+        # tries to be lenient in a similar way.
+        is_expected_status = (
+            response.status_code == HTTPStatus.OK or response.status_code == HTTPStatus.CREATED
+        )
+        try:
+            info = response.json()["trustedPhoneNumber"]
+        except KeyError:
+            if is_expected_status:
+                raise ITunesError(
+                    f"Trusted phone info missing from response with status: {response.status_code}"
+                )
+            else:
+                raise ITunesError(f"Unexpected response status: {response.status_code}")
+        else:
+            sentry_sdk.capture_message(
+                f"Unexpected response status from Apple ({response.status_code}) while requesting trusted phone info but found trustedPhoneNumber in response payload, proceeding using response's contents",
```

Comment:
I would advise against capturing this. If we don't care about the response status code, then let's just not check it. No point in keeping verbose code around for something we're not interested in.

Reply:
indirectly resolved by the addition of the `response.ok` check.